### PR TITLE
dm/openapi: support config experimental feature

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 

close https://github.com/pingcap/dm/issues/1982
and releated to https://github.com/pingcap/docs-dm/pull/812

### What is changed and how it works?

add `[experimental-features]` in dm-master config file

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release note 

```release-note
`support config experimental feature in dm-master config file`.
```

### CODE REVIEW

File: dm/dm/master/server.go [hunk 1]
```diff
@@ -184,15 +184,17 @@ func (s *Server) Start(ctx context.Context) (err error) {
...
-	initOpenAPIErr := s.InitOpenAPIHandles()
-	if initOpenAPIErr != nil {
-		return terror.ErrOpenAPICommonError.Delegate(initOpenAPIErr)
-	}
+
 	userHandles := map[string]http.Handler{
-		"/apis/":   apiHandler,
-		"/status":  getStatusHandle(),
-		"/debug/":  getDebugHandler(),
-		"/api/v1/": s.echo,
```
Comment (nongfushanquan) [inline dm/dm/master/server.go:195]: why handle this item in the different way?

Comment (Ehco1996) [reply dm/dm/master/server.go:195]: becasue all the openapi of dm is provided via `s.echo`, as this feature is experimental feature, we need handle this in different way

**Comment by sunzhaoyang** [comment]
LGTM

**Review by glorv — APPROVED** [review]

File: dm/dm/master/dm-master.toml [hunk 2]
```diff
@@ -39,5 +39,9 @@ rpc-timeout = "30s"
 # refilled at rate `rpc-rate-limit` tokens per second. Note `rpc-rate-limit`
 # is float64 type, so remember to add a decimal point and one trailing 0 if its
 # literal value happens to be an integer.
-rpc-rate-limit = 10.0
 rpc-rate-burst = 40
+rpc-rate-limit = 10.0
+
+# some experimental features
+[experimental-features]
```
Comment (lichunzhu) [inline dm/dm/master/dm-master.toml:46]: Just `experimental` to keep the same with tidb?
https://github.com/pingcap/tidb/blob/master/config/config.toml.example#L485

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

File: dm/dm/master/openapi_test.go [hunk 3]
```diff
@@ -129,6 +131,19 @@ func (t *openAPISuite) TestRedirectRequestToLeader(c *check.C) {
...
+	// create a new cluster
+	cfg1 := NewConfig()
+	c.Assert(cfg1.Parse([]string{"-config=./dm-master.toml"}), check.IsNil)
+	cfg1.Name = "dm-master-1"
+	cfg1.DataDir = c.MkDir()
+	cfg1.MasterAddr = tempurl.Alloc()[len("http://"):]
+	cfg1.PeerUrls = tempurl.Alloc()
+	cfg1.AdvertisePeerUrls = cfg1.PeerUrls
+	cfg1.InitialCluster = fmt.Sprintf("%s=%s", cfg1.Name, cfg1.AdvertisePeerUrls)
+	c.Assert(NewServer(cfg1).echo, check.IsNil)
```
Comment (lance6716) [inline dm/dm/master/openapi_test.go:144]: please make sure the old code will fail the test. You didn't `Start` so echo is always nil

**Review by lance6716 — CHANGES_REQUESTED** [review]
also prepare an "export for downgrade" config

---

update: check if Upgrade GitHub actions will pass

File: dm/dm/master/dm-master.toml (hunk 2 above)
Comment (Ehco1996) [reply dm/dm/master/dm-master.toml]: added in https://github.com/pingcap/ticdc/pull/3266/commits/cc7ef4d4e0e1c6d18049b71d5b7a1dc2064d0414

**Review by lance6716 — APPROVED** [review]
lgtm after we can confirm there's not too much trouble for downgrading from an OpenAPI enabled cluster

**Comment by Ehco1996** [comment]
> check if Upgrade GitHub actions will pass
upgrade CI passed in [here](https://github.com/Ehco1996/ticdc/actions/runs/1420325094) and i am doing downgrade test

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**